src/components/filters/DateFilter/DateFilter.tsx

```diff
@@ -30,12 +30,13 @@ interface IStatePropsIntersection {
     selectedFilterOption: ExtendedDateFilters.DateFilterOption;
 }
 
-export interface IDateFilterStateProps extends IStatePropsIntersection {
+export interface IDateFilterOwnProps extends IStatePropsIntersection {
     filterOptions: ExtendedDateFilters.IDateFilterOptionsByType;
     availableGranularities: ExtendedDateFilters.DateFilterGranularity[];
     isEditMode?: boolean;
     customFilterName?: string;
     dateFilterMode: ExtendedDateFilters.DashboardDateFilterConfigMode;
+    locale?: string;
```

Comment:
what will happen for invalid locales?

Reply:
You will got console error, plus there will be localization IDs used instead of translation.

Reply:
But there is note in documentation `The localization of the component. Defaults to `en-US`. For other languages, see the [full list of available localizations](https://github.com/gooddata/gooddata-react-components/tree/master/src/translations).`